runtime/jvmti/jvmtiRawMonitor.c

```diff
@@ -259,9 +259,43 @@ jvmtiRawMonitorWait(jvmtiEnv *env,
 			break;
 		case J9THREAD_INTERRUPTED:
 			/* FALLTHROUGH*/
-		case J9THREAD_PRIORITY_INTERRUPTED:
+		case J9THREAD_PRIORITY_INTERRUPTED: {
+#if JAVA_SPEC_VERSION >= 19
+			J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
+			j9object_t threadObject = NULL;
+			vmFuncs->internalEnterVMFromJNI(currentThread);
+			threadObject = currentThread->threadObject;
+			Assert_JVMTI_true(NULL != threadObject);
+			j9object_t threadLock = J9VMJAVALANGTHREAD_LOCK(currentThread, threadObject);
+			threadLock = (j9object_t)vmFuncs->objectMonitorEnter(currentThread, threadLock);
+			if (J9_OBJECT_MONITOR_ENTER_FAILED(threadLock)) {
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+				if (J9_OBJECT_MONITOR_CRIU_SINGLE_THREAD_MODE_THROW == (UDATA)threadLock) {
+					vmFuncs->setCRIUSingleThreadModeJVMCRIUException(currentThread, 0, 0);
+					rc = JVMTI_ERROR_THREAD_SUSPENDED;
+				} else
+#endif /* defined(J9VM_OPT_CRIU_SUPPORT) */
+				if (J9_OBJECT_MONITOR_OOM == (UDATA)threadLock) {
+					rc = JVMTI_ERROR_OUT_OF_MEMORY;
```

Comment:
the two error code set here will be overridden by line 296

Reply:
also note `JVMTI_ERROR_THREAD_SUSPENDED` is not a valid error code for this API

Reply:
> the two error code set here will be overridden by line 296

Fixed it.

> also note JVMTI_ERROR_THREAD_SUSPENDED is not a valid error code for this API

This error-check is CRIU specific; it won't affect normal API usage. It has been added at a few other places where object monitors are used. @tajila Do we need the above CRIU specific error check?

Reply:
@tajila @JasonFengJ9 Is the above CRIU specific change (ref: https://github.com/eclipse-openj9/openj9/pull/14651) valid for this scenario?

Reply:
Yes, we need the check here as well

Reply:
Updated to return `JVMTI_ERROR_INTERNAL`.